Intermittent integration test failure: test_first_run_release_date_visible_for_mobile[FIREFOX_IOS]

Seen here: https://app.circleci.com/pipelines/github/mozilla/experimenter/32613/workflows/30b99d66-bc3b-49b3-9a9c-4fb9718a3649/jobs/137685/steps

```py
_________ test_first_run_release_date_visible_for_mobile[FIREFOX_IOS] __________
[gw1] linux -- Python 3.10.12 /code/experimenter/tests/integration/.tox/integration-test-nimbus/bin/python

selenium = <selenium.webdriver.firefox.webdriver.WebDriver (session="d39918d5-a5ce-426f-a162-ec75f4a7c5af")>
kinto_client = <nimbus.kinto.client.KintoClient object at 0x7f05e177fd90>
application = 'IOS'
create_experiment = <function create_experiment.<locals>._create_experiment at 0x7f05e28a9ea0>
experiment_url = 'https://nginx/nimbus/test_first_run_release_date_visible_for_mobilefirefox_iosa1e7'

    @pytest.mark.nimbus_ui
    def test_first_run_release_date_visible_for_mobile(
        selenium,
        kinto_client,
        application,
        create_experiment,
        experiment_url,
    ):
        if application not in MOBILE_APPS:
            pytest.skip(f"Skipping for {application}")
    
        summary = create_experiment(selenium)
    
        audience = summary.navigate_to_audience()
        audience.make_first_run()
        audience.proposed_release_date = "2023-12-12"
    
        assert audience.is_first_run
        assert audience.proposed_release_date == "2023-12-12"
    
        audience.save_and_continue()
    
        summary = SummaryPage(selenium, experiment_url).open()
    
        assert summary.proposed_release_date == "2023-12-12"
    
>       summary.launch_and_approve()

nimbus/test_experimenter_ui.py:137: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
nimbus/pages/experimenter/summary.py:213: in launch_and_approve
    self.launch_without_preview.click()
nimbus/pages/experimenter/summary.py:184: in launch_without_preview
    return self.wait_for_and_find_element(*self._launch_without_preview_locator)
nimbus/pages/base.py:46: in wait_for_and_find_element
    self.wait_for_locator((strategy, locator), description)
nimbus/pages/base.py:40: in wait_for_locator
    self.wait.until(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <selenium.webdriver.support.wait.WebDriverWait (session="d39918d5-a5ce-426f-a162-ec75f4a7c5af")>
method = <selenium.webdriver.support.expected_conditions.presence_of_all_elements_located object at 0x7f05e177f1c0>
message = 'Summary Page: could not find locator'

    def until(self, method, message=''):
        """Calls the method provided with the driver as an argument until the \
        return value is not False."""
        screen = None
        stacktrace = None
    
        end_time = time.time() + self._timeout
        while True:
            try:
                value = method(self._driver)
                if value:
                    return value
            except self._ignored_exceptions as exc:
                screen = getattr(exc, 'screen', None)
                stacktrace = getattr(exc, 'stacktrace', None)
            time.sleep(self._poll)
            if time.time() > end_time:
                break
>       raise TimeoutException(message, screen, stacktrace)
E       selenium.common.exceptions.TimeoutException: Message: Summary Page: could not find locator

.tox/integration-test-nimbus/lib/python3.10/site-packages/selenium/webdriver/support/wait.py:80: TimeoutException
------------------------------- pytest-selenium --------------------------------
```

Some hints here:

![image](https://github.com/mozilla/experimenter/assets/119884/a9e105bc-bfb4-4578-96bd-b13b90207767)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-4135)

--- a/experimenter/tests/integration/nimbus/test_experimenter_ui.py
+++ b/experimenter/tests/integration/nimbus/test_experimenter_ui.py
@@ -128,9 +128,7 @@ def test_first_run_release_date_visible_for_mobile(
     assert audience.is_first_run
     assert audience.proposed_release_date == "2023-12-12"
 
-    audience.save_and_continue()
-
-    summary = SummaryPage(selenium, experiment_url).open()
+    summary = audience.save_and_continue()
 
     assert summary.proposed_release_date == "2023-12-12"
 

--- a/experimenter/tests/integration/nimbus/test_remote_settings.py
+++ b/experimenter/tests/integration/nimbus/test_remote_settings.py
@@ -234,12 +234,10 @@ def test_rollout_live_update_approve(
     audience = summary.navigate_to_audience()
 
     audience.percentage = "60"
-    audience.save_and_continue()
+    summary = audience.save_and_continue()
 
-    summary_page = SummaryPage(selenium, experiment_url).open()
-    summary_page.wait_for_update_request_visible()
-
-    summary_page.request_update_and_approve()
+    summary.wait_for_update_request_visible()
+    summary.request_update_and_approve()
     kinto_client.approve()
 
 
@@ -266,15 +264,14 @@ def test_rollout_live_update_approve_and_reject(
     audience = summary.navigate_to_audience()
 
     audience.percentage = "60"
-    audience.save_and_continue()
+    summary = audience.save_and_continue()
 
-    summary_page = SummaryPage(selenium, experiment_url).open()
-    summary_page.wait_for_update_request_visible()
+    summary.wait_for_update_request_visible()
 
-    summary_page.request_update_and_approve()
+    summary.request_update_and_approve()
     kinto_client.reject()
 
-    summary_page.wait_for_rejection_notice_visible()
+    summary.wait_for_rejection_notice_visible()
 
 
 @pytest.mark.remote_settings
@@ -300,17 +297,16 @@ def test_rollout_live_update_reject_on_experimenter(
     audience = summary.navigate_to_audience()
 
     audience.percentage = "60"
-    audience.save_and_continue()
+    summary = audience.save_and_continue()
 
-    summary_page = SummaryPage(selenium, experiment_url).open()
-    summary_page.wait_for_update_request_visible()
+    summary.wait_for_update_request_visible()
 
-    summary_page.request_update_and_reject()
-    summary_page.wait_for_rejection_reason_text_input_visible()
+    summary.request_update_and_reject()
+    summary.wait_for_rejection_reason_text_input_visible()
 
-    summary_page.set_rejection_reason()
-    summary_page.submit_rejection()
-    summary_page.wait_for_rejection_notice_visible()
+    summary.set_rejection_reason()
+    summary.submit_rejection()
+    summary.wait_for_rejection_notice_visible()
 
 
 @pytest.mark.remote_settings

